### PR TITLE
[Testing] Add more tests for ioThreadPolicies

### PR DESCRIPTION
**Pull request by stu-gott**
```release-note
NONE
```

### CODE REVIEW

File: tests/vmi_iothreads_test.go [hunk 1]
```diff
@@ -75,5 +79,148 @@ var _ = Describe("IOThreads", func() {
 			Expect(len(newVMI.Spec.Domain.Devices.Disks)).To(Equal(1))
 		})
 
+		It("[test_id:864][ref_id:2065] Should honor a mix of shared and dedicated ioThreadsPolicy", func() {
+			policy := v1.IOThreadsPolicyShared
+			vmi.Spec.Domain.IOThreadsPolicy = &policy
+
+			// The disk that came with the VMI
+			dedicated := true
+			vmi.Spec.Domain.Devices.Disks[0].DedicatedIOThread = &dedicated
```
Comment (sHaggYcaT) [inline tests/vmi_iothreads_test.go]: What's a point declare same variable here and in the 137 lines in the different contexts? Maybe better to declare it globally? And set it in the test cases how it requires?

File: tests/vmi_iothreads_test.go [hunk 2]
```diff
@@ -75,5 +79,148 @@ var _ = Describe("IOThreads", func() {
...
+			shr3, err := getDiskByName(domSpec, "shr2")
+			Expect(err).ToNot(HaveOccurred())
+			shr4, err := getDiskByName(domSpec, "shr2")
+			Expect(err).ToNot(HaveOccurred())
+
+			// the ioThreads for disks sh1 through shr4 will vary based on how many CPUs there are
+			// but we already verified the total number of threads, so we know they're spread out
+			// across the proper threadId pool.
+
+			By("Ensuring disk0 has a unique threadId")
```
Comment (sHaggYcaT) [inline tests/vmi_iothreads_test.go]: How do you think, maybe better to dump xml, unmarshall it, and parse it instead of use this function?

Comment (stu-gott) [reply tests/vmi_iothreads_test.go]: @sHaggYcaT It's honestly unclear what you're suggesting. lines 164 and 167 of this test are dumping and unmarshalling the domain XML:

```
By("Fetching the domain XML from the running pod")
domain, err := tests.GetRunningVirtualMachineInstanceDomainXML(virtClient, vmi)
Expect(err).ToNot(HaveOccurred())
domSpec := &api.DomainSpec{}
Expect(xml.Unmarshal([]byte(domain), domSpec)).To(Succeed())
```

So I guess the part that's unclear to me is when you suggest "and parse it". What exactly are you picturing?

File: tests/vmi_iothreads_test.go (hunk 1 above)
Comment (stu-gott) [reply tests/vmi_iothreads_test.go]: fixed thanks!

File: tests/vmi_iothreads_test.go (hunk 2 above)
Comment (sHaggYcaT) [reply tests/vmi_iothreads_test.go]: Yeah, you're right @stu-gott

**Comment by sHaggYcaT** [comment]
lgtm

**Pull request merged**